norman_utils_external/secure_random_int_generator.py

```diff
@@ -2,6 +2,49 @@
 
 
 class SecureRandomIntGenerator:
+    """
+    Cryptographically secure random integer generator for producing uniformly
+    distributed integers within a specified inclusive range.
+
+    This generator uses `SecureRandomBytesGenerator` (ChaCha20-based) as its
+    entropy source and applies *rejection sampling* to guarantee an unbiased
+    distribution even when the range does not evenly divide the underlying
+    byte space.
+
+    **How It Works**
+
+    - A stream of secure random bytes is generated using ChaCha20.
+    - The random bytes are converted to an integer (`value`).
+    - Values falling inside a “rejection zone”—which would cause modulo bias—
+      are discarded.
+    - The accepted value is reduced to the desired range and returned.
+
+    **Attributes**
```

Comment:
Remove. **attributes** refers to class attributes which are stable across releases, not internal implementation variables which are amenable to change at any commit.